cranelift/codegen/src/isa/aarch64/abi.rs

```diff
@@ -135,12 +135,20 @@ impl Into<AMode> for StackAMode {
 // Returns the size of stack space needed to store the
 // `int_reg` and `vec_reg`.
 fn saved_reg_stack_size(
+    call_conv: isa::CallConv,
     int_reg: &[Writable<RealReg>],
     vec_reg: &[Writable<RealReg>],
 ) -> (usize, usize) {
     // Round up to multiple of 2, to keep 16-byte stack alignment.
     let int_save_bytes = (int_reg.len() + (int_reg.len() & 1)) * 8;
-    let vec_save_bytes = vec_reg.len() * 16;
+    let vec_reg_size = if call_conv.extends_baldrdash() { 16 } else { 8 };
```

Comment:
A comment here to explain the strategies for both cases would make it a bit clearer, I think -- something like "The Baldrdash ABI saves whole (16-byte) clobbered vector registers, so there is no need to worry about 16-byte stack alignment. The SystemV (AAPCS) ABI saves half (8-byte) clobbered vector registers, so we round up if odd."?